projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts

```diff
@@ -21,38 +21,20 @@ export class MerchandisingCarouselComponent {
     protected componentData: CmsComponentData<
       CmsMerchandisingCarouselComponent
     >,
-    protected merchandisingCarouselComponentService: MerchandisingCarouselComponentService
+    protected merchandisingCarouselComponentService: MerchandisingCarouselComponentService,
+    protected el: ElementRef
   ) {}
 
   private componentData$: Observable<
     CmsMerchandisingCarouselComponent
   > = this.componentData.data$.pipe(filter(Boolean));
 
-  carouselStyle$: Observable<{
-    [key: string]: string;
-  }> = this.componentData$.pipe(
-    map(data => {
-      const style = {};
-      if (data.backgroundColour) {
-        style['background-color'] = data.backgroundColour;
-      }
-      if (data.textColour) {
-        style['color'] = data.textColour;
-      }
-      return style;
-    })
+  backgroundColor$: Observable<string> = this.componentData$.pipe(
```

Comment:
I'd simplify the code as mentioned at the html. Do some operations on the (various?)  streams, and set the variables during the operation.